test/rules/deprecated_member_use_from_same_package_test.dart

```diff
@@ -0,0 +1,786 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:collection/collection.dart';
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../rule_test_support.dart';
+
+main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(DeprecatedMemberUseFromSamePackageTest);
+  });
+}
+
+@reflectiveTest
+class DeprecatedMemberUseFromSamePackageTest extends LintRuleTest {
+  @override
+  String get lintRule => 'deprecated_member_use_from_same_package';
+
+  @override
+  Future<void> assertDiagnostics(
+      String code, List<ExpectedDiagnostic> expectedDiagnostics) async {
+    addTestFile(code);
+    await resolveTestFile();
+    var filteredErrors = errors
+        .whereNot((e) =>
```

Comment:
I'm curious what errors you're filtering out here? Could we re-write this to be more targeted? (My concern is that we not ignore/mask surprising errors in the test data.) FWIW I'm all for enumerating a handful of diagnostics that might be ignored in a lot of tests.  (I was thinking just today how we might want to ignore `unused_local_variable` at least.)

Reply:
These are targeted at the two error codes listed. What are you thinking for "more targeted"? I'm open to ideas here, cuz it's a little weird. I can just add a comment too.

Reply:
The filter removed everything except the listed codes. I think the question is: what codes are being removed? In general we try to write code that doesn't produce any diagnostics other than the ones we're explicitly testing for. That helps us catch code that needs to be updated as the language changes, but ignoring everything negates that benefit.

Reply:
I might be thinking about this wrong, but normally we take _all_ reported errors and compare them against the short list of expected errors.

> I think the question is: what codes are being removed?

In this bit, I am taking all errors which are not `HintCode.DEPRECATED_MEMBER_USE_FROM_SAME_PACKAGE` or `HintCode.DEPRECATED_MEMBER_USE_FROM_SAME_PACKAGE_WITH_MESSAGE`. So it's the same set of reported errors as usual, minus these two.

> ignoring everything negates that benefit.

I certainly don't intend to ignore everything. Just two error codes.

These two error codes are produced today, but will not be produced soon, with an SDK commit. The only way to get the tests to pass in google3 in both states is to filter out the HintCodes.

Reply:
I read the code backwards and thought you were filtering out everything except those two. Filtering those two is fine.